hack/make/build-deb

```diff
@@ -38,7 +39,7 @@ set -e
 	# TODO decide if it's worth getting go-md2man in _each_ builder environment to avoid this
 
 	# TODO add a configurable knob for _which_ debs to build so we don't have to modify the file or build all of them every time we need to test
-	for dir in contrib/builder/deb/*/; do
+	for dir in contrib/builder/deb/${DOCKER_ENGINE_GOARCH:="amd64"}/*/; do
```

Comment:
Now as the `.detect-daemon-osarch` is available and the `DOCKER_ENGINE_GOARCH` is always set we could remove the default values here. Sounds good.